frontend/public/components/factory/list-page.jsx

```diff
@@ -57,6 +59,14 @@ TextFilter.displayName = 'TextFilter';
 
 /** @augments {React.PureComponent<{ListComponent: React.ComponentType<any>, kinds: string[], flatten?: function, data?: any[], rowFilters?: any[]}>} */
 export class ListPageWrapper_ extends React.PureComponent {
+  componentDidMount() {
+    if (!_.isEmpty(this.props.kinds) && _.size(this.props.kinds) === 1) {
```

Comment:
You can remove the isEmpty check since you're already checking if size if 1 (which will be false if empty)